### PR TITLE
FS-2103 Secure commenter routes

### PR DESCRIPTION
**Pull request by harryyo**
### Remove scoring features from assessment criteria and tasklist views for Commenters

- Return 404 from sub criteria page when thene_id=score and user is not Assessor or Lead Assessor
- Add conditional logic to remove total scores and scores columns from tasklist view for Commenters
- Add conditional logic to remove Score the criteria link from sub-criteria navbar for Commenters
- Add Flask g variable to global template constants for easy use within templates
- Add tests to check above Commenter functionality
- Updated default DEBUG_USER_ROLE to LEAD_ASSESSOR so that full functionality is visible in development by default

Also small fix to add int filter to state.funding_amount_requested formatter in assessor_tasklist.html to ensure correct data type is passed to numeric format function

### CODE REVIEW

File: app/assess/routes.py [hunk 1]
```diff
@@ -28,89 +26,76 @@
...
-        "is_commenter": g.user.highest_role == "COMMENTER",
-        "scoring_permissions": any(
-            required_role in g.user.roles
-            for required_role in ["ASSESSOR", "LEAD_ASSESSOR"]
-        ),
-    }
-    
-    displayCommentBox = False
-    if 'add-comment' in request.args.keys() and request.args['add-comment'] == '1':
-        displayCommentBox = True
```
Comment (tferns) [inline app/assess/routes.py:49]: Is this being unnecessarily removed?

Comment (tferns) [reply app/assess/routes.py:49]: Maybe a botched merge conflict resolution?

File: app/assess/routes.py [hunk 2]
```diff
@@ -28,89 +26,76 @@
...
-                    sub_criteria_id=sub_criteria_id,
-                    user_id=user_id,
-                    theme_id=theme_id
-                )
-            return redirect(request.path + "?theme_id=" + theme_id)
-        else:
-            score_error = True if not form.score.data else False
-            justification_error = (
-                True if not form.justification.data else False
-            )
```
Comment (tferns) [inline app/assess/routes.py:92]: Also looks significantly different

File: tests/api_data/endpoint_data.json [hunk 3]
```diff
@@ -529,21 +529,21 @@
             "id": "benefits",
```
Comment (tferns) [inline tests/api_data/endpoint_data.json:529]: Are these changes correct?

File: tests/test_routes.py [hunk 4]
```diff
@@ -235,3 +242,80 @@ def test_route_sub_criteria_side_bar(
...
+            "app.assess.models.ui.applicants_response.get_file_url",
+            return_value="sample1.doc",
+        )
+        # Mocking fsd-user-token cookie
+        test_payload = {
+            "accountId": "test-user",
+            "email": "test@example.com",
+            "fullName": "Test User",
+            "roles": ["COMMENTER"],
+        }
```
Comment (tferns) [inline tests/test_routes.py:297]: Can we use the `test_commenter_claims` constant so we're not repeating this? 

https://github.com/communitiesuk/funding-service-design-assessment/blob/037a5d69c15a808a3e9a23186c0ebc73b49e2264/tests/conftest.py#L31-L36

File: tests/test_routes.py [hunk 5]
```diff
@@ -235,3 +242,80 @@ def test_route_sub_criteria_side_bar(
...
+        self, flask_test_client
+    ):
+
+        # Mocking fsd-user-token cookie
+        test_payload = {
+            "accountId": "test-user",
+            "email": "test@example.com",
+            "fullName": "Test User",
+            "roles": ["COMMENTER"],
+        }
```
Comment (tferns) [inline tests/test_routes.py:256]: Can we use the `test_commenter_claims` constant so we're not repeating this? 

https://github.com/communitiesuk/funding-service-design-assessment/blob/037a5d69c15a808a3e9a23186c0ebc73b49e2264/tests/conftest.py#L31-L36

**Review by tferns — CHANGES_REQUESTED** [review]
Refactor looks good, but I think there's been a botched resolution of merge conflicts as changes from https://github.com/communitiesuk/funding-service-design-assessment/pull/92 looks to be missing

File: app/assess/routes.py (hunk 1 above)
Comment (harryyo) [reply app/assess/routes.py:49]: role_information dict has been replaced as flask g is now available globally as a template variable

Comment (harryyo) [reply app/assess/routes.py:49]: display_comment_box functionality has been simplified and moved to logic on line 113-134

File: app/assess/routes.py (hunk 2 above)
Comment (harryyo) [reply app/assess/routes.py:92]: comment_form.validate_on_submit() block has been simplified and moved to line 138

Comment (harryyo) [reply app/assess/routes.py:92]: Scoring post logic has been moved to line 78 inside `if theme_id == "score"` block starting on line 57

File: tests/api_data/endpoint_data.json (hunk 3 above)
Comment (harryyo) [reply tests/api_data/endpoint_data.json:529]: Yep the assessment_store endpoint responds with uppercase status enums with _ afaik

File: tests/test_routes.py (hunk 4 above)
Comment (harryyo) [reply tests/test_routes.py]: Yep done

**Review by tferns — APPROVED** [review]

**Pull request merged**